helix-term/src/commands.rs

```diff
@@ -1800,7 +1800,9 @@ fn search_selection(cx: &mut Context) {
         .join("|");
 
     let msg = format!("register '{}' set to '{}'", '/', &regex);
-    cx.editor.registers.get_mut('/').push(regex);
+    if let Some(r) = cx.editor.registers.get_mut('/') {
+        r.push(regex);
+    }
```

Comment:
If we change this to write then we lose the search history.
When you yank to the search register, all the previous search history is deleted, since yank uses write instead of push.

Reply:
Ah right, I missed that. Scratch the part about `vec![regex]`. I think it should be possible to eliminate this `Registers::get_mut` though if we introduce a `Registers::push`

Reply:
Yes, i made those changes and pushed it.